storage/range.go

```diff
@@ -819,21 +864,9 @@ func (r *Range) applyRaftCommand(ctx context.Context, index uint64, originNodeID
 			if log.V(1) {
 				log.Infoc(ctx, "found response cache entry for %+v", args.Header().CmdID)
 			}
-			return reply.Header().GoError()
-		} else if ok && err != nil {
-			newErr := util.Errorf("unable to read result for %+v from the response cache: %s", args, err)
-			reply.Header().SetGoError(newErr)
-			return newErr
-		}
-	}
-
-	// Verify the leader lease is held; Note that we don't require the
```

Comment:
The old code checked the response cache before the leader lease on purpose: if there is an entry in the response cache then this is a retry and the lease was valid at the time the command was executed, so we can return the result. With this change it is possible for a client to submit a command, lose its connection (so it doesn't see that the command succeed), then retry and see an error. Of course, the NotLeaderError is retryable so the client will just try again on the new leader and should eventually get through, so I think all this really saves us is an additional retry on rare cases that are already failing and retrying.